cmd/kubeadm/app/cmd/config.go

```diff
@@ -502,13 +504,30 @@ type ImagesList struct {
 	cfg *kubeadmapi.InitConfiguration
 }
 
+// ConfigImagesList represents information for JSON and YAML output produced by 'kubeadm config images list'
+type ConfigImagesList struct {
+	Images []string
+}
+
 // Run runs the images command and writes the result to the io.Writer passed in
-func (i *ImagesList) Run(out io.Writer) error {
-	imgs := images.GetControlPlaneImages(&i.cfg.ClusterConfiguration)
-	for _, img := range imgs {
-		fmt.Fprintln(out, img)
+func (i *ImagesList) Run(out io.Writer, outputFormat string) error {
+	of := strings.ToLower(outputFormat)
+	if of != "text" && of != "json" && of != "yaml" {
+		return errors.Errorf("invalid output format: %s", outputFormat)
```

Comment:
this check seems to be present both here and in MashalTo.

Reply:
ok, will try to remove it from here.